### DISCUSSION TITLE
Controlling the music assistant volume

### DISCUSSION THREAD

**Discussion by mikebakke** [music-assistant/support#2330] [Q&A] [answered]
I have installed the add-on into HAOS and it's all working fine. The only issue is that I'm casting to Nest mini devices and when I set automations to increase/decrease volume it goes in steps of 10. This seems to be because the nest does it that way when I touch the top or click + or - on the device in HA. 

Thing is that if I open the active player in MASS I can drag the vol slider in increments of 1 and that reflects correctly on the playing device. What I have been unable to do so far is control the MASS volume in HA - I don't see an entrity for that incremental player volume slider. Is it exposed somewhere?

**Comment by OzGav** [comment]
You just use the HA media player service calls. One is `media_player.volume_set`

**Reply by mikebakke** [reply]
Yes - I was using music player: turn up/down volume service calls - I had seen the set volume call but I saw that ranges from 0 to 1 and there is no existing MASS variable/sensor I could see to increment or decrement hence being stuck with the nest "jumps" in volume levels.

I guess I'll have to try and figure out how to create a persistent variable to work from unless you think it's something that can be added in a future release :-)

Thanks

**Reply by OzGav** [reply]
You can call volume_set a MA playerwith any number from 0 to 100?

**Reply by mikebakke** [reply]
I assumed that was possible, the thing is that I can't read the current value. I have a remote Ikea button = single click down, double up - to use set I need the current value to add then set. This is why I think I need to create a persistent variable in HA. I will try to figure that out and see what happens.

Cheers

**Reply by OzGav** [reply, marked as answer]
Yes using an input number which you increase or decrease by one by a pair of automations and then a third automation which sets the player volume whenever the input number changes
